File: windows/Sources/JazzCaptureCore/ContinuousCapture.cs
namespace JazzCaptureCore;

/// <summary>Continuous-mode decisions shared by startup, Pause/Resume and session rollover.</summary>
public static class ContinuousCapture
{
    public static readonly TimeSpan PauseReminderInterval = TimeSpan.FromMinutes(30);

    public static bool ShouldRemind(bool enabled, bool paused, bool recording) =>
        enabled && paused && !recording;

    // A failed commit is never permission to overwrite the engine that owns its journal.
    public static bool ShouldContinue(bool enabled, bool committed, bool paused, bool terminating) =>
        enabled && committed && !paused && !terminating;
}

File: macos/Sources/JazzCaptureCore/ContinuousCapture.swift
import Foundation

/// Continuous-mode lifecycle rules; the native hosts own their timers and capture resources.
public enum ContinuousCapture {
    public static let pauseReminderInterval: TimeInterval = 30 * 60

    public static func shouldRemind(enabled: Bool, paused: Bool, recording: Bool) -> Bool {
        enabled && paused && !recording
    }

    public static func shouldContinue(
        enabled: Bool, committed: Bool, paused: Bool, terminating: Bool
    ) -> Bool {
        enabled && committed && !paused && !terminating
    }
}
